tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/Proxy/InferredProxyCoordinator.cs

```diff
@@ -0,0 +1,75 @@
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+#nullable enable
+
+using System;
+using Datadog.Trace.Logging;
+using Datadog.Trace.Propagators;
+
+namespace Datadog.Trace.ClrProfiler.AutoInstrumentation.Proxy;
+
+/// <summary>
+/// Coordinates the extraction of proxy metadata and creation of proxy spans.
+/// Acts as the main entry point for creating inferred proxy spans.
+/// </summary>
+internal class InferredProxyCoordinator
+{
+    private static readonly IDatadogLogger Log = DatadogLogging.GetLoggerFor<InferredProxyCoordinator>();
+    private readonly IInferredSpanFactory _spanFactory;
+    private readonly IInferredProxyExtractor _extractor;
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="InferredProxyCoordinator"/> class.
+    /// </summary>
+    /// <param name="extractor">Component that extracts proxy metadata from headers</param>
+    /// <param name="spanFactory">Component that creates spans from the extracted metadata</param>
+    public InferredProxyCoordinator(IInferredProxyExtractor extractor, IInferredSpanFactory spanFactory)
+    {
+        _spanFactory = spanFactory;
+        _extractor = extractor;
+    }
+
+    /// <summary>
+    /// Extracts proxy metadata from headers and creates a corresponding span.
+    /// </summary>
+    /// <returns>
+    /// When successful, returns an object containing:
+    /// <list type="bullet">
+    /// <item>The created <see cref="Scope"/> containing the proxy <see cref="Span"/></item>
+    /// <item>An updated <see cref="PropagationContext"/> that includes the new <see cref="SpanContext"/> and previous <see cref="Baggage"/>.</item>
+    /// </list>
+    /// Returns <see langword="null"/> if extraction fails or span creation fails.
+    /// </returns>
+    public InferredProxyScopePropagationContext? ExtractAndCreateScope<TCarrier, TCarrierGetter>(
+        Tracer tracer,
+        TCarrier carrier,
+        TCarrierGetter carrierGetter,
+        PropagationContext propagationContext)
+        where TCarrierGetter : struct, ICarrierGetter<TCarrier>
+    {
+        try
+        {
+            if (!_extractor.TryExtract(carrier, carrierGetter, tracer, out var proxyData))
+            {
+                return null;
+            }
+
+            var scope = _spanFactory.CreateSpan(tracer, proxyData, propagationContext.SpanContext);
+            if (scope == null)
+            {
+                return null;
+            }
+
+            var updatedContext = new PropagationContext(scope.Span.Context, propagationContext.Baggage); // TODO is this correct?
+            return new InferredProxyScopePropagationContext(scope, updatedContext);
+        }
+        catch (Exception ex)
+        {
+            Log.Error(ex, "Error processing inferred proxy span."); // TODO, we have too many Log.Errors, we should consider removing this one as the other two log errors
```

Comment:
```suggestion
```

`TryExtract` has a `catch` and will log
`CreateSpan` has a `catch` and will log

I don't think anything else realistically will get here, but if so we will already have an error log

Or we can just remove the todo

```suggestion
            Log.Error(ex, "Error processing inferred proxy span.");
```

Reply:
I left the extra try/catch since it doesn't hurt (and to be extra safe). I removed the TODO comment.